pkg/kudoctl/util/repo/repo_framework.go

```diff
@@ -0,0 +1,211 @@
+	if err != nil {
+		return nil, errors.Wrap(err, "parsing config url")
+	}
+	parsedURL.Path = fmt.Sprintf("%s/index.yaml", strings.TrimSuffix(parsedURL.Path, "/"))
+
+	indexURL = parsedURL.String()
+
+	resp, err := r.Client.Get(indexURL)
+	if err != nil {
+		return nil, errors.Wrap(err, "getting index url")
+	}
+
+	indexBytes, err := ioutil.ReadAll(resp)
+	if err != nil {
+		return nil, errors.Wrap(err, "reading index response")
+	}
+
+	indexFile, err := parseIndexFile(indexBytes)
+	return indexFile, err
+}
+
+// DownloadBundle downloads the tgz file from the given repo
+func (r *FrameworkRepository) DownloadBundle(bundleName string) (*FrameworkBundle, error) {
+	var fileURL string
+	parsedURL, err := url.Parse(r.Config.URL)
+	if err != nil {
+		return nil, errors.Wrap(err, "parsing config url")
+	}
+	parsedURL.Path = fmt.Sprintf("%s/%s.tgz", parsedURL.Path, bundleName)
+
+	fileURL = parsedURL.String()
+
+	resp, err := r.Client.Get(fileURL)
+	if err != nil {
+		return nil, errors.Wrap(err, "getting file url")
+	}
+
+	bundle, err := untar(resp)
+
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed unpacking %s", bundleName)
+	}
+
+	return bundle, nil
+}
+
+// GetFrameworkVersionDependencies returns a slice of strings that contains the names of all dependency Frameworks
+func (r *FrameworkRepository) GetFrameworkVersionDependencies(name string, fv *v1alpha1.FrameworkVersion) ([]string, error) {
+	var dependencyFrameworks []string
+	if fv.Spec.Dependencies != nil {
+		for _, v := range fv.Spec.Dependencies {
+			dependencyFrameworks = append(dependencyFrameworks, v.Name)
+		}
+	}
+	return dependencyFrameworks, nil
+}
+
+func untar(r io.Reader) (*FrameworkBundle, error) {
+
+	gzr, err := gzip.NewReader(r)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		err := gzr.Close()
+		if err != nil {
+			fmt.Printf("Error when closing gzip reader %s", err)
+		}
+	}()
+
+	tr := tar.NewReader(gzr)
+
+	result := &FrameworkBundle{}
+	for {
+		header, err := tr.Next()
+
+		switch {
+
+		// if no more files are found return
+		case err == io.EOF:
+			if result.Instance != nil && result.FrameworkVersion != nil && result.Framework != nil {
+				// bundle is complete
+				return result, nil
+			}
+
+			var missing []string
+			if result.Instance == nil {
+				missing = append(missing, "instance.yaml")
+			} else if result.FrameworkVersion != nil {
+				missing = append(missing, "frameworkversion.yaml")
+			} else if result.Framework != nil {
+				missing = append(missing, "framework.yaml")
+			}
+			return nil, fmt.Errorf("incomplete bundle - these files are missing: %v", missing)
+
+		// return any other error
+		case err != nil:
+			return nil, err
+
+		// if the header is nil, just skip it (not sure how this happens)
+		case header == nil:
+			continue
+		}
+
+		// check the file type
+		switch header.Typeflag {
+
+		case tar.TypeDir:
+			// we don't handle folders right now, the structure is flat
+
+		// if it's a file create it
```

Comment:
this is actually not creating files anymore. `// if its a file unmarshall it and add to result` ?